src/Microsoft.DotNet.Build.CloudTestTasks/ValidateWorkItems.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    public class ValidateWorkItems : Task
+    {
+        private List<ITaskItem> processedWorkItems = new List<ITaskItem>();
+
+        /// <summary>
+        /// Helix work items to be validated.
+        /// Checks include making sure minimum metadata is present and that the path is relative to the supplied root path.
+        /// </summary>
+        public ITaskItem[] WorkItems { get; set; }
+
+        [Output]
+        public ITaskItem [] ProcessedWorkItems { get; set; }
+
+        /// <summary>
+        /// While we're validating, may as well calculate the relative blob path
+        /// Since we'd also like to be able to check that the path makes sense.
+        /// Ignore this check if the item already has this metadata.
+        /// </summary>
+        public String WorkItemArchiveRoot { get; set; }
+
+        private readonly string[] requiredMetadataFields = { "TimeoutInSeconds", "Command", "PayloadFile", "TimeoutInSeconds" };
+
+        public override bool Execute()
+        {
+            //bool isTrue = true;
+            //Log.LogMessage(MessageImportance.High, "Please attach");
+            //while (isTrue) { };
```

Comment:
oops, not supposed to be here.  I find my silly approach to not end up creating excess instances of VS :)